hyperspy/_signals/lazy.py

```diff
@@ -525,83 +532,109 @@ def _map_iterate(self,
-            raise ValueError('"ragged" kwarg has to be bool for lazy signals')
-        _logger.debug("Entering '_map_iterate'")
-
-        size = max(1, self.axes_manager.navigation_size)
-        from hyperspy.misc.utils import (create_map_objects,
-                                         map_result_construction)
-        func, iterators = create_map_objects(function, size, iterating_kwargs,
-                                             **kwargs)
-        iterators = (self._iterate_signal(), ) + iterators
-        res_shape = self.axes_manager._navigation_shape_in_array
-        # no navigation
-        if not len(res_shape) and ragged:
-            res_shape = (1,)
-
-        all_delayed = [dd(func)(data) for data in zip(*iterators)]
-
-        if ragged:
-            if inplace:
-                raise ValueError("In place computation is not compatible with "
-                                  "ragged array for lazy signal.")
-            # Shape of the signal dimension will change for the each nav.
-            # index, which means we can't predict the shape and the dtype needs
-            # to be python object to support numpy ragged array
-            sig_shape = ()
-            sig_dtype = np.dtype('O')
+        """This function has two different implementations
+        based on if there is a BaseSignal passed as an argument
+        or not.
+        """
+        # unpacking keyword arguments
+        if ragged and inplace:
+            raise ValueError("Ragged and inplace are not compatible with a lazy signal")
+        autodetermine = (output_signal_size is None or output_dtype is None)
+
+        rechunked_signals = ()
+        iter_keys = []
+        testing_kwargs = {}
+        nav_indexes = self.axes_manager.navigation_indices_in_array
+        if len(iterating_kwargs) > 0:
+            for key, signal in iterating_kwargs:
+                if (signal.axes_manager.navigation_shape != self.axes_manager.navigation_shape and
+                        signal.axes_manager.navigation_shape != ()):
+                    if signal.axes_manager.signal_shape == ():
+                        kwargs[key] = signal.data  # this really isn't an iterating signal.
+                    else:
+                        raise ValueError('the size of the navigation_shape: <' +
+                                        str(signal.axes_manager.navigation_shape) +
+                                        '> must be consistent with the size of the mapped signal <' +
+                                        str(self.axes_manager.navigation_shape) + '>')
+                elif signal.axes_manager.navigation_shape == ():
+                    if signal.axes_manager.signal_shape == (1,):
+                        kwargs[key] = signal.data[0]  # this really isn't an iterating signal.
+                    else:
+                        kwargs[key] = signal.data
+                else:
+                    rechunked_signal, test_kwarg = rechunk_signal(signal,
```

Comment:
So the reason for all of this rechunking is for the case of some signal iterating alongside some other signal.  Those two signals have to have the same chunk structure in the navigation dimension as each other or else the map blocks function doesn't work.

If that chunk strucuture is already there I don't think it really matters if you rechunk as it shouldn't really take any time

Reply:
I see. The thing is that HyperSpy chunks always contain at least one full signal in that chunk. That is fine when the signal is small, but for large signals (e.g HR images) and suitable operations it may be better to use a different chunking style. Currently, that's not possible because of the automatic re-chunking inside map. Alternatively:

1. the user could disable chunking and take the responsibility of verifying that all signals are chunk-compatible
2.``map`` could offer the option to rechunk only the "keyword signals" to make them compatible with the main signal